### PR TITLE
GN4 mapper: Map keyword types/themes into a new Keyword type

### PR DESCRIPTION
**Pull request by tkohr**
### Description

This PR takes the keywords from ES `allKeywords` instead of `tags` in order to map the keyword type/theme like `place`, `theme` etc. into a new `Keyword` type.

### Quality Assurance Checklist

- [x] Commit history is devoid of any _merge commits_ and readable to facilitate reviews
- [x] If **new logic** ⚙️ is introduced: unit tests were added
- [ ] If **new user stories** 🤏 are introduced: E2E tests were added
- [ ] If **new UI components** 🕹️ are introduced: corresponding stories in Storybook were created
- [ ] If **breaking changes** 🪚 are introduced: add the `breaking change` label
- [ ] If **bugs** 🐞 are fixed: add the `backport <release branch>` label
- [ ] The [documentation website](docs) 📚 has received the love it deserves

**This work is sponsored by [MEL](https://www.lillemetropole.fr/)**.

### CODE REVIEW

File: libs/api/metadata-converter/src/lib/gn4/atomic-operations.ts [hunk 1]
```diff
@@ -98,3 +100,19 @@ export const mapContact = (
...
+
+export const mapKeywords = (thesauri: Thesaurus[], language: string) => {
+  const keywords = []
+  for (const thesaurusId in thesauri) {
+    const thesaurus = thesauri[thesaurusId]
+    for (const keyword of thesaurus.keywords) {
+      keywords.push({
+        label: selectTranslatedValue<string>(keyword, language),
+        type: getKeywordTypeFromKeywordTypeCode(thesaurus.theme),
+        ...(thesaurus.id && { thesaurus: { id: thesaurus.id } }),
```
Comment (tkohr) [inline libs/api/metadata-converter/src/lib/gn4/atomic-operations.ts:112]: Just wondering if we shouldn't try to recover the `name` and `url` here as well? I see optionally `title` and `link` in the ES response.

**Review by tkohr — COMMENTED** [review]
Thanks for the huge work behind this @jahow ! As far as I can judge, all looks good. I just left one comment and added one small commit for the sake of renaming completeness.

File: libs/api/metadata-converter/src/lib/gn4/atomic-operations.ts (hunk 1 above)
Comment (jahow) [reply libs/api/metadata-converter/src/lib/gn4/atomic-operations.ts:112]: yes good point!

Comment (jahow) [reply libs/api/metadata-converter/src/lib/gn4/atomic-operations.ts]: done

**Review by jahow — APPROVED** [review]
Looking good :) thanks for starting this work, now we have proper keywords in the pivot format!

**Comment by tkohr** [comment]
Thanks for the latest adaption @jahow ! Looking at some other code a last doubt came to my mind, but I guess we can still continue using `tag` for the keywords filter [here](https://github.com/geonetwork/geonetwork-ui/blob/place-keywords/libs/feature/search/src/lib/utils/service/fields.service.ts#L63), right?

**Comment by jahow** [comment]
> Thanks for the latest adaption @jahow ! Looking at some other code a last doubt came to my mind, but I guess we can still continue using `tag` for the keywords filter [here](https://github.com/geonetwork/geonetwork-ui/blob/place-keywords/libs/feature/search/src/lib/utils/service/fields.service.ts#L63), right?

Yes that should keep working identically

**Pull request merged**